api/nodes/views.py

```diff
@@ -303,6 +303,16 @@ def get_queryset(self):
                     raise PermissionDenied
             return nodes
         else:
+            if self.request.query_params.get('filter[preprint]'):
+                preprints = True if self.request.query_params.get('filter[preprint]') == 'true' else False
+                query = self.get_query_from_request()
+                if preprints:
+                    nodes = Node.find(query)
+                    return [node for node in nodes if node.is_preprint]
```

Comment:
Prefer generators over list comprehensions so that you don't have to evaluate every node before returning: 
`(node for node in nodes if node.is_preprint)`